src/wallet/spend.cpp

```diff
@@ -1251,8 +1251,9 @@ static util::Result<CreatedTransactionResult> CreateTransactionInternal(
 
     if (gArgs.GetBoolArg("-walletrejectlongchains", DEFAULT_WALLET_REJECT_LONG_CHAINS)) {
         // Lastly, ensure this tx will pass the mempool's chain limits
-        if (!wallet.chain().checkChainLimits(tx)) {
-            return util::Error{_("Transaction has too long of a mempool chain")};
+        auto result = wallet.chain().checkChainLimits(tx);
+        if (!result) {
+            return util::Error{util::ErrorString(result)};
```

Comment:
Just a comment: Would be nice to get #25665 merged, so we can get a move constructor here.

Reply:
Thanks @TheCharlatan that can come as a follow-up post https://github.com/bitcoin/bitcoin/pull/25665 I think?

Reply:
Indeed, or it could even be done as part of that PR. It depends on which one gets merged first.